scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/AvroBucketMetadata.java

```diff
@@ -173,6 +175,12 @@ private static String validateKeyField(String keyField, Class keyClass, Schema s
   }
 
   private static Class<?> getKeyClassFromSchema(Schema schema) {
+    Schema.Type schemaType = getSchemaTypeForUnionType(schema);
+
+    return getClassForType(schemaType);
+  }
+
+  private static Schema.Type getSchemaTypeForUnionType(Schema schema) {
```

Comment:
```suggestion
  private static Schema.Type getSchemaTypeForUnionType(Schema schema) {
  private static Schema.Type getSchemaOrInnerUnionType(Schema schema) {
```
I think this method name is a bit clearer, wdyt?